pkg/ssh/utils.go

```diff
@@ -21,6 +21,7 @@ func Validate(user *url.Userinfo, path string, port int, identity string) (*conf
 	if strings.Contains(path, "/run") {
```

Comment:
Non-blocking: BTW naming a transformative parser like this “validate” is a bit misleading.

---

Non-blocking: And why are native callers passing `options.Identity` to _this_ function, and immediately afterwards to `ValidateAndConfigure`? It seems to me that this could be simplified, e.g. by this function returning a struct (`resolvedConnectionInfo`[1]?), and callers can then pass around that _single_ struct and access `dest.URI` and `dest.Identity`.

[1] I guess that struct can’t be `*config.Destination`, because the on-disk format and the normalized internal data seem to differ.

Reply:
(Regardless of the name, AFAICS `Validate` does not need to be public at all; so, with the recent plans to make the c/common API stable, it should start private. We can easily make things public later, but we can’t take back public API nearly as easily. This applies to other functions in this package as well.)